stdlib/public/core/CString.swift

```diff
@@ -124,6 +124,40 @@ extension String {
     self = ""
   }
 
+  /// Creates a new string by copying and validating the null-terminated UTF-8
+  /// data referenced by the given pointer.
+  ///
+  /// This initializer does not try to repair ill-formed UTF-8 code unit
+  /// sequences. If any are found, the result of the initializer is `nil`.
+  ///
+  /// The following example calls this initializer with pointers to the
+  /// contents of two different `CChar` arrays---the first with well-formed
+  /// UTF-8 code unit sequences and the second with an ill-formed sequence at
+  /// the end.
+  ///
+  ///     let validUTF8: [CChar] = [67, 97, 102, -61, -87, 0]
+  ///     validUTF8.withUnsafeBufferPointer { ptr in
+  ///         let s = String(validatingCString: ptr.baseAddress!)
+  ///         print(s)
+  ///     }
+  ///     // Prints "Optional("Café")"
+  ///
+  ///     let invalidUTF8: [CChar] = [67, 97, 102, -61, 0]
+  ///     invalidUTF8.withUnsafeBufferPointer { ptr in
+  ///         let s = String(validatingCString: ptr.baseAddress!)
+  ///         print(s)
+  ///     }
+  ///     // Prints "nil"
+  ///
+  /// - Parameter nullTerminatedUTF8:
+  ///       A pointer to a null-terminated UTF-8 code sequence.
+  @inlinable
+  @_alwaysEmitIntoClient
+  public init?(validatingCString nullTerminatedUTF8: UnsafePointer<CChar>) {
+    // FIXME: https://github.com/apple/swift/issues/68433 (rdar://115296219)
+    self.init(validatingUTF8: nullTerminatedUTF8)
+  }
```

Comment:
It would seem preferable to point the ABI entry point to `String.init?(validatingCString:)`, but renaming initializers isn't supported by `@_silgen_name`. This solution will cause a warning when compiling the stdlib in Swift 6.